prac_07/convert_miles_km.py

```diff
@@ -0,0 +1,42 @@
+"""
+CP1404 Practical
+GUI program to convert miles to kilometres
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+
+
+__author__ = 'Dea Roys'
+
+MILES_TO_KM = 1.60934
+
+class Miles_To_Km(App):
+
+    def build(self):
+        self.title = "Convert miles to kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def check_inputs(self):
+        try:
+            user_input = float(self.root.ids.input_number.text)
+            return user_input
+        except ValueError:
+            return 0
+
+    def handle_increment(self,value):
+        """Increments the user input by 1 or -1."""
+        user_input = self.check_inputs()
+        incremented_value = user_input + value
+        self.root.ids.input_number.text = str(incremented_value)
+        self.calculation()
+
+    def calculation(self):
+        """Convert miles to km."""
+        check_number = self.check_inputs()
+        miles_in_km = check_number * MILES_TO_KM
+        self.root.ids.output_number.text = str(miles_in_km)
```

Comment:
Has the calculation already been performed or does this method **calculate** something? Where does this value to be calculated come from? Is this a flexible method? 

Does the text location have a **Property** which it can use to update itself? Is miles_in_km a necessary variable or could the calculation be incorporated _into_ the value assignment somehow?